File: law-service/src/main/java/avg/vnlaw/lawservice/entities/Article.java
package avg.vnlaw.lawservice.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;

@Entity
@Table(name ="pdarticle")
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Builder
public class Article extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String name;

    @Column(nullable = false, columnDefinition = "LONGTEXT")
    private String content;

    @Column(name = "`index`", nullable = false)
    private String index;

    @Column(nullable = false,columnDefinition = "TEXT")
    private String vbqppl;

    @Column(name= "`vbqppl_link`",nullable = true,columnDefinition = "TEXT")
    private String vbqpplLink;

    @Column(name = "`order`", nullable = false)
    private Integer order;

    @Column(name="effective_date", nullable = false)
    private Date effectiveDate;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "topic_id", nullable = false)
    private Topic topic;

    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subject_id",nullable = false)
    private Subject subject;

    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    @ManyToOne(fetch =  FetchType.LAZY)
    @JoinColumn(name = "chapter_id",nullable = false)
    private Chapter chapter;

}

File: front-end/mobile/lib/data/models/vbqppl/article.dart
class Article {
  final String id;
  final String name;
  final String content;
  final String index;
  final int? order;
  final String vbqppl;
  final String vbqpplLink;
  final List<dynamic> tables;
  final List<dynamic> files;

  Article({
    required this.id,
    required this.name,
    required this.content,
    required this.index,
    this.order,
    required this.vbqppl,
    required this.vbqpplLink,
    required this.tables,
    required this.files,
  });

  factory Article.fromJson(Map<String, dynamic> json) {
    return Article(
      id: json['id'],
      name: json['name'],
      content: json['content'] ?? '',
      index: json['index'] ?? '',
      order: json['order'],
      vbqppl: json['vbqppl'] ?? '',
      vbqpplLink: json['vbqpplLink'] ?? '',
      tables: json['tables'] ?? [],
      files: json['files'] ?? [],
    );
  }

  String getSummary() {
    if (content.length > 600) {
      return '${content.substring(0, 600)}...';
    }
    return content;
  }
}
